app/src/main/java/com/example/triptracker/viewmodel/UserProfileViewModel.kt

```diff
@@ -53,63 +53,29 @@ class UserProfileViewModel(
-    userProfileRepository.updateUserProfile(updatedProfile)
-
-    val updatedFollowerProfile = follower.copy(following = follower.following + userProfile.mail)
-    userProfileRepository.updateUserProfile(updatedFollowerProfile)
-  }
-
-  /**
-   * This function adds new following to the user profile in the database.
-   *
-   * @param userProfile : user profile to update
-   * @param following : following to add
+   * @param userProfile : the user profile to which we add a follower
+   * @param follower : the user profile of the follower to add
    */
-  fun addFollowingInDb(userProfile: UserProfile, following: UserProfile) {
-    val updatedProfile = userProfile.copy(following = userProfile.following + following.mail)
-    userProfileRepository.updateUserProfile(updatedProfile)
-
-    val updatedFollowingProfile = following.copy(followers = following.followers + userProfile.mail)
-    userProfileRepository.updateUserProfile(updatedFollowingProfile)
+  fun addFollower(userProfile: UserProfile, follower: UserProfile) {
+    val updatedUserProfile = userProfile.copy(followers = userProfile.followers + follower.mail)
+    val updatedFollower = follower.copy(following = follower.following + userProfile.mail)
+    userProfileRepository.updateUserProfile(updatedUserProfile)
+    userProfileRepository.updateUserProfile(updatedFollower)
   }
 
   /**
-   * This function removes a follower from the user profile in the database.
+   * Function that remove the follower from the user profile
    *
-   * @param userProfile : user profile to update
-   * @param follower : follower to remove
+   * @param userProfile : the user profile from which we remove a follower
+   * @param follower : the user profile of the follower to remove
    */
-  fun removeFollowerInDb(userProfile: UserProfile, follower: UserProfile) {
-    val updatedProfile =
-        userProfile.copy(followers = userProfile.followers.filter { it != follower.mail })
-    userProfileRepository.updateUserProfile(updatedProfile)
-
-    val updatedFollowerProfile =
-        follower.copy(following = follower.following.filter { it != userProfile.mail })
-    userProfileRepository.updateUserProfile(updatedFollowerProfile)
-  }
-
-  /**
-   * This function removes a following from the user profile in the database.
-   *
-   * @param userProfile : user profile to update
-   * @param following : following to remove
-   */
-  fun removeFollowingInDb(userProfile: UserProfile, following: UserProfile) {
-    val updatedProfile =
-        userProfile.copy(following = userProfile.following.filter { it != following.mail })
-    userProfileRepository.updateUserProfile(updatedProfile)
-
-    val updatedFollowingProfile =
-        following.copy(followers = following.followers.filter { it != userProfile.mail })
-    userProfileRepository.updateUserProfile(updatedFollowingProfile)
+  fun removeFollower(userProfile: UserProfile, follower: UserProfile) {
```

Comment:
Good change this avoids a lot of useless code and doesn't affect any functionality fo the app